allennlp/models/coreference_resolution/coref.py

```diff
@@ -252,6 +264,9 @@ def forward(self,  # type: ignore
         output_dict = {"top_spans": top_spans,
                        "antecedent_indices": valid_antecedent_indices,
                        "predicted_antecedents": predicted_antecedents}
+        if keep_antecedent_alternatives and keep_antecedent_alternatives[0]:
```

Comment:
Why is `if keep_antecedent_alternatives` not enough? The implicit `bool` check should handle `None` and having members simultaneously:
```python
In [284]: l1 = None

In [285]: l2 = []

In [286]: l3 = [2]

In [287]: if l1:
     ...:     print("apple")
     ...: elif l2:
     ...:     print("banana")
     ...: elif l3:
     ...:     print("pear")
     ...:
pear
```

Reply:
But `bool([False])` is `True`, which is the situation here.  (one the initial instance, you have a field `keep_antecedent_alternatives` which maps to a boolean value, but the vectorization code in AllenNLP wraps the boolean value in a list by the time it gets here

Reply:
Ah, I missed "containing the single value ``True``".

Reply:
But that does remind me that this checks only if the first thing in a batch has requested coref_scores, and we want to return them if *any* do. Just forced-pushed an amended commit with a fix.